### PR TITLE
Fix conditions to apply common flags

### PR DESCRIPTION
**Pull request by tytan652**
When I tried to build it on my Arch Linux the flag `-fPIC` was not applied so I investigated the issue.

Inside [CommonFlags.cmake](https://github.com/aja-video/ntv2/blob/abf17cc1e7aadd9f3e4972774a3aba2812c51b75/cmake/CommonFlags.cmake) many conditions are not working, so most of the flags are not set.

I just tried to fix them by using regex, if you want me to replace it by `OR` and `STREQUAL` just ask me.

### CODE REVIEW

**Comment by paulh-aja** [comment]
@tytan652 These regex conditions don't seem to work for me on my Mac. Did you test the regex as working on a particular system? If I revert back to the original syntax it does work on my Mac and Windows machines. I'm not sure where I picked up the syntax I initially used here, or if it is even actually correct. Trying to do some digging in the CMake docs...

**Comment by paulh-aja** [comment]
https://cmake.org/cmake/help/latest/command/string.html#regex-specification

Looking at the CMake Docs it seems like the correct regex MATCHES expression should be something like this:
```
elseif(CMAKE_CXX_COMPILER_ID MATCHES "[Clang|GNU]")
    if (CMAKE_SYSTEM_NAME MATCHES "[Linux|Darwin]")
```

What do you think @tytan652 ?

File: cmake/CommonFlags.cmake [hunk 1]
```diff
@@ -3,20 +3,20 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL MSVC)
...
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang|GNU)
-    if (CMAKE_SYSTEM_NAME STREQUAL Linux|Darwin)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "^(Clang|GNU)$")
+    if (CMAKE_SYSTEM_NAME MATCHES "^(Linux|Darwin)$")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
     endif()
 
-    if(CMAKE_BUILD_TYPE STREQUAL Debug|RelWithDebInfo)
+    if(CMAKE_BUILD_TYPE MATCHES "(^Debug|RelWithDebInfo)$")
```
Comment (paulh-aja) [inline cmake/CommonFlags.cmake:19]: I think this might be a typo?
Should be:
`if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")`

**Comment by paulh-aja** [comment]
At first this change wasn't working properly for me on my Mac but I think I've identified a small typo in one of the regex conditions.
`if(CMAKE_BUILD_TYPE MATCHES "(^Debug|RelWithDebInfo)$")`
 should be changed to:
`if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")`

Does that look correct to you @tytan652 ?

**Comment by tytan652** [comment]
> ```
> elseif(CMAKE_CXX_COMPILER_ID MATCHES "[Clang|GNU]")
>     if (CMAKE_SYSTEM_NAME MATCHES "[Linux|Darwin]")
> ```
This ask for a string containing at least those letters `x`,`i`,`n`,`u`,`L` or `w` `a` `r` `D` `i`.
Letter are not put in order to indicate that it does not match words but letters.

File: cmake/CommonFlags.cmake (hunk 1 above)
Comment (tytan652) [reply cmake/CommonFlags.cmake:19]: It's a typo

Comment (tytan652) [reply cmake/CommonFlags.cmake:19]: I'll push in a minute

**Comment by paulh-aja** [comment]
Ah, I see another reason why it's not working on my Mac. The CMAKE_CXX_COMPILER_ID variable is returning "AppleClang".

**Comment by paulh-aja** [comment]
> > ```
> > elseif(CMAKE_CXX_COMPILER_ID MATCHES "[Clang|GNU]")
> >     if (CMAKE_SYSTEM_NAME MATCHES "[Linux|Darwin]")
> > ```
> 
> This ask for a string containing at least those letters `x`,`i`,`n`,`u`,`L` or `w` `a` `r` `D` `i`. Letter are not put in order to indicate that it does not match words but letters with any length limit.

You're right, after re-reading it I see that I was mistaken. I think the main problems now are the small typo which you just fixed, and then the fact that CMAKE_CXX_COMPILER_ID returns "AppleClang" on macOS. When I'd originally created these conditions I was primarily testing on Windows. I'd never gotten around to testing these statements more carefully on Mac and Linux until now.

**Comment by tytan652** [comment]
I can add `AppleClang` to other GNU-Clang regex.

**Review by paulh-aja — APPROVED** [review]
Looks good now. I will merge the PR.

Thank you so much, @tytan652.

**Pull request merged**